python/ray/data/datasource/partitioning.py

```diff
@@ -402,6 +402,7 @@ def _parse_dir_path(self, dir_path: str) -> Dict[str, str]:
         return {field_names[i]: d for i, d in enumerate(dirs)} if dirs else {}
 
 
+@DeveloperAPI
```

Comment:
This is used in the `read_*()` [APIs](https://github.com/ray-project/ray/blob/52b1e4b6722ff681f536e8df3a12754b6018e702/python/ray/data/read_api.py#L347); should this be a public API?

Reply:
Changed to PublicAPI("beta")